src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java

```diff
@@ -119,6 +120,41 @@
                 .build());
     }
 
+    @PostMapping(path = "/consent-in-contested/send-order", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "'Consent Order Approved' callback handler for consent in contested. Checks state and if "
+        + "approved generates docs else puts latest general order into uploadORder fields. Then sends the data to bulk print")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Callback was processed successfully or in case of an error message is attached to the case",
+            response = AboutToStartOrSubmitCallbackResponse.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Internal Server Error")
+    })
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> consentInContestedSendOrder(
+        @RequestHeader(value = AUTHORIZATION_HEADER) String authToken,
+        @NotNull @RequestBody @ApiParam("CaseData") CallbackRequest callback) throws JsonProcessingException {
+        CaseDetails caseDetails = callback.getCaseDetails();
+        Map<String, Object> caseData = caseDetails.getData();
+
+        if (caseDetails.getState().equals(CONSENTED_ORDER_APPROVED)) {
+            CaseDocument orderLetter = consentOrderApprovedDocumentService.generateApprovedConsentOrderLetter(caseDetails, authToken);
```

Comment:
this is a lot to put in a controller, can we move all of this to the `consentOrderApprovedDocumentService` and just call when the above if statement is true please